critiquebrainz/frontend/external/spotify.py

```diff
@@ -6,12 +6,57 @@
 import requests
 import urllib.parse
 from brainzutils import cache
+from base64 import b64encode
+from flask import current_app as app
+from critiquebrainz.frontend.external.exceptions import SpotifyWebAPIException
 
 DEFAULT_CACHE_EXPIRATION = 12 * 60 * 60  # seconds (12 hours)
+ACCESS_TOKEN_EXPIRATION = 60 * 60 # seconds (1 hour)
 
 BASE_URL = "https://api.spotify.com/v1"
 
 
+def _fetch_access_token():
+    """Get an access token from the oauth credentials."""
+
+    key = cache.gen_key("spotify_access_token")
+    namespace = "spotify_access_token"
+    access_token = cache.get(key, namespace)
+
+    client_id = app.config.get("SPOTIFY_CLIENT_ID")
+    client_secret = app.config.get("SPOTIFY_CLIENT_SECRET")
+    auth_value = b64encode(bytes(f"{client_id}:{client_secret}", "utf-8")).decode("utf-8")
+
+    if not access_token:
+        access_token = requests.post(
+            "https://accounts.spotify.com/api/token",
+            data={"grant_type": "client_credentials"},
+            headers={"Authorization": f"Basic {auth_value}"},
+        ).json()
+        access_token = access_token.get('access_token')
+        if not access_token:
+            raise SpotifyWebAPIException("Could not fetch access token for Spotify API")
+        cache.set(key=key, namespace=namespace, val=access_token, time=ACCESS_TOKEN_EXPIRATION)
+    return access_token
+
+
+def _get_spotify(query):
+    """Make a GET request to Spotify Web API.
+
+    Args:
+        query (str): Query to the Web API.
+
+    Returns:
+        Dictionary containing the information.
+    """
+    access_token = _fetch_access_token()
+    url = BASE_URL + query
+    headers = {"Authorization": f"Bearer {access_token}"}
+
+    result = requests.get(f"{url}", headers=headers)
```

Comment:
You are right, we don't need it.

And no, we haven't discussed this. I think f-strings are new in Python 3, so we couldn't have used it before. This inconsistency might be partially my fault since at the beginning I used `%` and then started switching to `.format` since it allows to specify placeholder names explicitly.